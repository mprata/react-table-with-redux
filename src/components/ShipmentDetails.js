import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchShipment } from "../actions";

import CargoDetails from "./CargoDetails";
import SerivcesDetails from "./ServicesDetails";
import NameForm from "./NameForm";

class ShipmentDetails extends React.Component {
    componentDidMount() {
        const { id } = this.props.match.params;
        if (this.props.shipment.length === 0) {
            this.props.fetchShipment(id);
        }
    }
    render() {
        if (!this.props.shipment.id) {
            return <div></div>
        }
        return (
            <div className="ui container">
                <div className="ui top attached header">
                    <Link to="/shipments">Back</Link>
                </div>
                <h5 className="ui attached header">ID</h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.id}</p>
                </div>
                <h5 className="ui attached header">Name</h5>
                <div className="ui attached segment">
                    <NameForm />
                </div>
                <h5 className="ui attached header">Cargo Details</h5>
                <div className="ui attached segment">
                    <CargoDetails cargo={this.props.shipment.cargo} />
                </div>
                <h5 className="ui attached header">Mode</h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.mode}</p>
                </div>
                <h5 className="ui attached header">
                    Type
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.type}</p>
                </div>
                <h5 className="ui attached header">
                    Destination
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.destination}</p>
                </div>
                <h5 className="ui attached header">
                    Origin
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.origin}</p>
                </div>
                <h5 className="ui attached header">
                    Services
                    </h5>
                <div className="ui attached segment">
                    <SerivcesDetails services={this.props.shipment.services} />
                </div>
                <h5 className="ui attached header">
                    Total
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.total}</p>
                </div>
                <h5 className="ui attached header">
                    Status
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.status}</p>
                </div>
                <h5 className="ui attached header">
                    User ID
                    </h5>
                <div className="ui attached segment">
                    <p>{this.props.shipment.userId}</p>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return { shipment: state.selectedShipment }
}

export default connect(mapStateToProps, { fetchShipment })(ShipmentDetails);