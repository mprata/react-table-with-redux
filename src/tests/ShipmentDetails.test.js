import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from 'react-redux';
import ShipmentDetails from "../components/ShipmentDetails";
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';

const initialState = {};
const createMockStore = configureMockStore(initialState);
const props = {
    selectedShipment: {
        "id": "S1000",
        "name": "My First",
        "cargo": [
            {
                "type": "Fabric",
                "description": "1000 Blue T-shirts",
                "volume": "2"
            },
            {
                "type": "Fabric",
                "description": "2000 Green T-shirts",
                "volume": "3"
            }
        ],
        "mode": "sea",
        "type": "FCL",
        "destination": "SaarbrÃ¼cker Str. 38, 10405 Berlin",
        "origin": "Shanghai Port",
        "services": [
            {
                "type": "customs"
            }
        ],
        "total": "1000",
        "status": "ACTIVE",
        "userId": "U1000"
    }
};
const mockStore = createMockStore(props);

const routeParams = {
    match: {
        params: {
            id:"S1000"
        }
    }
};

it('ShipmentDetails component should render', () => {
    const component = renderer.create(
        <Provider store={mockStore}>
            <Router><ShipmentDetails {...routeParams}/></Router>
        </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Check dom structure of shipment detail component, if the page is created correctly', () => {
    const { getByText } = render(
        <Provider store={mockStore}>
            <Router><ShipmentDetails {...routeParams}/></Router>
        </Provider>
    );

    expect(getByText(/Back/i)).toBeTruthy();
    expect(getByText(/S1000/i)).toBeTruthy();
    expect(getByText(/Update/i)).toBeTruthy();
    cleanup();
});