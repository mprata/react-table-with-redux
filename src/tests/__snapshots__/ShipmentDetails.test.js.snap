// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipmentDetails component should render 1`] = `
<div
  className="ui container"
>
  <div
    className="ui top attached header"
  >
    <a
      href="/shipments"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
  <h5
    className="ui attached header"
  >
    ID
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      S1000
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Name
  </h5>
  <div
    className="ui attached segment"
  >
    <form
      className="ui form error"
      onSubmit={[Function]}
    >
      <div
        className="field "
      >
        <input
          autoComplete="off"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
      <button
        className="ui button primary"
      >
        Update
      </button>
    </form>
  </div>
  <h5
    className="ui attached header"
  >
    Cargo Details
  </h5>
  <div
    className="ui attached segment"
  >
    <div>
      <table
        className="ui collapsing table"
      >
        <thead>
          <tr>
            <th>
              Type
            </th>
            <th>
              Description
            </th>
            <th>
              Volume
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Fabric
            </td>
            <td>
              1000 Blue T-shirts
            </td>
            <td>
              2
            </td>
          </tr>
          <tr>
            <td>
              Fabric
            </td>
            <td>
              2000 Green T-shirts
            </td>
            <td>
              3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <h5
    className="ui attached header"
  >
    Mode
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      sea
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Type
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      FCL
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Destination
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      Saarbr√ºcker Str. 38, 10405 Berlin
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Origin
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      Shanghai Port
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Services
  </h5>
  <div
    className="ui attached segment"
  >
    <div>
      <table
        className="ui collapsing table"
      >
        <thead>
          <tr>
            <th>
              Type
            </th>
            <th>
              Value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              customs
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <h5
    className="ui attached header"
  >
    Total
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      1000
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    Status
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      ACTIVE
    </p>
  </div>
  <h5
    className="ui attached header"
  >
    User ID
  </h5>
  <div
    className="ui attached segment"
  >
    <p>
      U1000
    </p>
  </div>
</div>
`;
